### ISSUE TITLE
ALOps Extension API - Release pipeline succeeds but deployment failed

### ISSUE THREAD

**Issue by JeppeBylov** [HodorNV/ALOps#205]
**Describe the bug**
I have a very simple release pipeline, with only 1 step - ALOps Extension API.
The latest artifact from a build pipeline is released to a SaaS sandbox. The deployment happens, but the installation fails due to an error regarding applicationarea property missing.
BUT the pipeline gets status succeeded.

**the used yaml**
please provide the yaml that you used.  It helps you put the yaml like this: 
```yaml
#Your build pipeline references a secret variable named ‘bcpassword’. Create or edit the build pipeline for this YAML file, define the variable on the Variables tab, and then select the option to make it secret. See https://go.microsoft.com/fwlink/?linkid=865972
variables:
  azure_tenant_id: 'vwsf.onmicrosoft.com'
  bcusername: 'VWSF JOB QUEUE'

steps:
- task: Hodor.hodor-alops.ALOpsExtensionAPI.ALOpsExtensionAPI@1
  displayName: 'ALOps Extension API'
  inputs:
    interaction: publish
    api_endpoint: 'https://api.businesscentral.dynamics.com/v2.0/$(azure_tenant_id)/QA/api'
    authentication: basic
    azure_tenant_id: '$(azure_tenant_id)'
    username: '$(bcusername)'
    password: '$(bcpassword)'

```

**the output**
Also the complete output is necessary for us to see what is going on.  Also use backtics:
```
2020-09-27T07:45:11.5816485Z ##[section]Starting: ALOps Extension API
2020-09-27T07:45:11.5951371Z ==============================================================================
2020-09-27T07:45:11.5951697Z Task         : ALOps Extension API
2020-09-27T07:45:11.5951951Z Description  : Get/Publish extensions with the Business Central API
2020-09-27T07:45:11.5952193Z Version      : 1.434.1755
2020-09-27T07:45:11.5952368Z Author       : Hodor
2020-09-27T07:45:11.5953308Z Help         : Get/Publish extensions with the Business Central API.
2020-09-27T07:45:11.5953671Z ==============================================================================
2020-09-27T07:45:13.3533611Z *** Validate configuration
2020-09-27T07:45:13.9867754Z 
2020-09-27T07:45:13.9881946Z *** No ALOps License found. Set License with the [alops-licenseid] or [alops_licenseid] variables.
2020-09-27T07:45:13.9893110Z *** To acquire a license, please visit: https://www.alops.be/getlicense
2020-09-27T07:45:13.9907668Z *** For pricing options, please visit : https://www.alops.be/pricing
2020-09-27T07:45:13.9920366Z *** For documentation, please visit   : https://www.alops.be/documentation
2020-09-27T07:45:13.9925238Z 
2020-09-27T07:45:15.2192854Z *** Imported required PS-Functions
2020-09-27T07:45:15.2209585Z *** Parse Task Parameters
2020-09-27T07:45:15.2644956Z *** Creating API Connector with endpoint [https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/QA/api]
2020-09-27T07:45:15.2807428Z *** Interaction Mode: publish
2020-09-27T07:45:15.2840547Z *** API Authentication: Using Basic Authentication [VWSF JOB QUEUE]
2020-09-27T07:45:15.2913099Z *** List Companies
2020-09-27T07:45:15.2983495Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies'
2020-09-27T07:45:16.3960576Z 
2020-09-27T07:45:16.3990622Z 
2020-09-27T07:45:16.4022906Z id                : 0123f70d-92f2-ea11-aa62-0022489b1355
2020-09-27T07:45:16.4025022Z systemVersion     : 16.5.15897.16301
2020-09-27T07:45:16.4027078Z name              : 1_Master Company- No Posting
2020-09-27T07:45:16.4028575Z displayName       : 1_Master Company- No Posting
2020-09-27T07:45:16.4029439Z businessProfileId : 
2020-09-27T07:45:16.4030197Z 
2020-09-27T07:45:16.4056623Z 
2020-09-27T07:45:16.4080082Z 
2020-09-27T07:45:16.4123847Z *** Selected Company: 1_Master Company- No Posting [0123f70d-92f2-ea11-aa62-0022489b1355]
2020-09-27T07:45:16.4217736Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensions'
2020-09-27T07:45:16.5897673Z *** Installed Extensions:
2020-09-27T07:45:16.6009433Z 
2020-09-27T07:45:16.6123800Z publisher         displayName                                         version id                                  
2020-09-27T07:45:16.6132736Z ---------         -----------                                         ------- --                                  
2020-09-27T07:45:16.6140315Z Continia Software Continia Core                                       3.0     4b915d7e-c02a-435f-85ab-649086c1e002
2020-09-27T07:45:16.6150594Z Continia Software Continia Document Capture 365                       6.5     6da8dd2f-e698-461f-9147-8e404244dd85
2020-09-27T07:45:16.6157973Z Continia Software Continia Expense Management 365                     6.5     8d4eab29-8c7f-4b6c-be9a-b7fdfb9da196
2020-09-27T07:45:16.6164954Z Microsoft         AMC Banking 365 Fundamentals                        16.5    16319982-4995-4fb1-8fb2-2b1e13773e3b
2020-09-27T07:45:16.6172714Z Microsoft         Application                                         16.5    c1335042-3002-4257-bf8a-75c898ccb1b8
2020-09-27T07:45:16.6181345Z Microsoft         Base Application                                    16.5    437dbf0e-84ff-417a-965d-ed2bb9650972
2020-09-27T07:45:16.6189006Z Microsoft         Business Central Cloud Migration - Previous Release 16.5    6992416f-3f39-4d3c-8242-3fff61350bea
2020-09-27T07:45:16.6196839Z Microsoft         Business Central Intelligent Cloud                  16.5    334ef79e-547e-4631-8ba1-7a7f18e14de6
2020-09-27T07:45:16.6203907Z Microsoft         C5 2012 Data Migration                              16.5    3d5b2137-eeeb-4014-8489-41d37f8fd4c3
2020-09-27T07:45:16.6211532Z Microsoft         Essential Business Headlines                        16.5    2a89f298-7ffd-44a5-a7ce-e08dac98abce
2020-09-27T07:45:16.6217870Z Microsoft         Intelligent Cloud Base                              16.5    58623bfa-0559-4bc2-ae1c-0979c29fd9e0
2020-09-27T07:45:16.6227106Z Microsoft         Late Payment Prediction                             16.5    3d5b2137-efeb-4014-8489-41d37f8fd4c3
2020-09-27T07:45:16.6234236Z Microsoft         OIOUBL                                              16.5    ac762be1-e90f-4a72-b519-612a5e3ddc2e
2020-09-27T07:45:16.6241375Z Microsoft         Payment and Reconciliation Formats (DK)             16.5    644f7e5b-a81c-408a-aaa2-766e655a80a3
2020-09-27T07:45:16.6248367Z Microsoft         PayPal Payments Standard                            16.5    d09fa965-9a2a-424d-b704-69f3b54ed0ce
2020-09-27T07:45:16.6255480Z Microsoft         Payroll Data Import Definitions (DK)                16.5    a3c8a6e2-834a-4e73-b1f1-79e76b652659
2020-09-27T07:45:16.6263024Z Microsoft         Sales and Inventory Forecast                        16.5    c526b3e9-b8ca-4683-81ba-fcd5f6b1472a
2020-09-27T07:45:16.6270308Z Microsoft         Send To Email Printer                               16.5    8c972578-fe72-4aa5-ae51-cc5575fef2ea
2020-09-27T07:45:16.6277207Z Microsoft         System Application                                  16.5    63ca2fa4-4f03-4f2b-a480-172fef340d3f
2020-09-27T07:45:16.6283874Z Microsoft         Tax File Formats (DK)                               16.5    2c97db9b-4aef-41e2-aa3e-03fa892c6815
2020-09-27T07:45:16.6290222Z Microsoft         VAT Group Management                                16.5    c50a4bf0-db51-4ad2-88d5-fe2287da0eb8
2020-09-27T07:45:16.6296953Z Microsoft         WorldPay Payments Standard                          16.5    bae453ed-0fd8-4416-afdc-4b09db6c12c3
2020-09-27T07:45:16.6297560Z 
2020-09-27T07:45:16.6298148Z 
2020-09-27T07:45:16.6304603Z 
2020-09-27T07:45:16.6316089Z *** Published Extensions (not installed):
2020-09-27T07:45:16.6345397Z 
2020-09-27T07:45:16.6376044Z *** Import NAV/BC Management DLL's
2020-09-27T07:45:16.6403047Z *** ServiceTier not specified, loading default assemblies.
2020-09-27T07:45:16.6549173Z *** Selected module: [C:\Program Files\Microsoft Dynamics NAV\150\Service\Microsoft.Dynamics.Nav.Apps.Management.dll]
2020-09-27T07:45:16.6562724Z *** Loading module: [C:\Program Files\Microsoft Dynamics NAV\150\Service\Microsoft.Dynamics.Nav.Apps.Management.dll]
2020-09-27T07:45:17.6679126Z *** Selected module: [C:\Program Files\Microsoft Dynamics NAV\150\Service\Microsoft.Dynamics.Nav.Management.dll]
2020-09-27T07:45:17.6690549Z *** Loading module: [C:\Program Files\Microsoft Dynamics NAV\150\Service\Microsoft.Dynamics.Nav.Management.dll]
2020-09-27T07:45:17.7639721Z *** Publish Extension
2020-09-27T07:45:17.7650976Z   * Artifact Path  : C:\Agent\vwsfagent\_work\r1\a
2020-09-27T07:45:17.7661751Z   * Artifact Filter: *.app
2020-09-27T07:45:17.7773803Z *** Resolve App file in [C:\Agent\vwsfagent\_work\r1\a]
2020-09-27T07:45:17.7890485Z *** Resolved App File: [Default publisher_BQS_1.0.38.0.app].
2020-09-27T07:45:17.7914476Z *** Resolved App Path: [C:\Agent\vwsfagent\_work\r1\a\_BQS\VWSF\Default publisher_BQS_1.0.38.0.app]
2020-09-27T07:45:17.7977474Z *** Upload extension [C:\Agent\vwsfagent\_work\r1\a\_BQS\VWSF\Default publisher_BQS_1.0.38.0.app]
2020-09-27T07:45:17.7991017Z *** Get App Info
2020-09-27T07:45:19.1833068Z  * App.ID        = 6672e461-6086-456a-83dd-14ffc34020f8
2020-09-27T07:45:19.1866422Z  * App.Name      = BQS
2020-09-27T07:45:19.1892978Z  * App.Publisher = Default publisher
2020-09-27T07:45:19.1921704Z  * App.Version   = 1.0.38.0
2020-09-27T07:45:19.1933950Z *** Get Extension status
2020-09-27T07:45:19.1950652Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensions'
2020-09-27T07:45:19.3922686Z *** Get Extension Deployment status
2020-09-27T07:45:19.3962409Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:45:19.5748459Z ##[command]Invoke-RestMethod -Method Patch -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionUpload(0)/content'
2020-09-27T07:45:21.0744966Z 
2020-09-27T07:45:21.0762379Z *** Poll Deployment status
2020-09-27T07:45:21.0786180Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:45:26.3641564Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:45:31.6144860Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:45:36.8987469Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:45:42.1682530Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:45:47.4256939Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:45:52.7167780Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:45:57.9920228Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:46:03.2113346Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:46:08.3981398Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:46:13.6153787Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:46:18.8438711Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:46:24.0918157Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:46:29.3391231Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:46:29.4860383Z *** Deployment Status:
2020-09-27T07:46:29.4877797Z 
2020-09-27T07:46:29.4880862Z 
2020-09-27T07:46:29.4887000Z operationID : 16acc1bd-c300-4699-be44-c4be7466d20c
2020-09-27T07:46:29.4888909Z name        : BQS
2020-09-27T07:46:29.4890700Z publisher   : Default publisher
2020-09-27T07:46:29.4892210Z appVersion  : 1.0.38.0
2020-09-27T07:46:29.4894243Z status      : Failed
2020-09-27T07:46:29.4895973Z startedOn   : 2020-09-27T07:45:21.033Z
2020-09-27T07:46:29.4897031Z 
2020-09-27T07:46:29.4898043Z 
2020-09-27T07:46:29.4899764Z 
2020-09-27T07:46:29.4943176Z ##[error]App publish failed. Check the 'Extension Deployment Status' page in the Business Central client.
2020-09-27T07:46:29.4953083Z ##[command]Invoke-RestMethod -Method Get -Uri 'https://api.businesscentral.dynamics.com/v2.0/vwsf.onmicrosoft.com/qa/api/microsoft/automation/beta/companies(0123f70d-92f2-ea11-aa62-0022489b1355)/extensionDeploymentStatus'
2020-09-27T07:46:29.5609282Z 
2020-09-27T07:46:29.5647831Z name                            publisher         appVersion operationType schedule  status    startedOn               
2020-09-27T07:46:29.5662203Z ----                            ---------         ---------- ------------- --------  ------    ---------               
2020-09-27T07:46:29.5672021Z BQS                             Default publisher 1.0.38.0   Upload        Immediate Failed    2020-09-27T07:45:21.033Z
2020-09-27T07:46:29.5682087Z Continia Document Capture 365   Continia Software            Install                 Completed 2020-09-14T08:18:48.687Z
2020-09-27T07:46:29.5691917Z Continia Expense Management 365 Continia Software            Install                 Completed 2020-09-14T08:22:21.27Z 
2020-09-27T07:46:29.5692495Z 
2020-09-27T07:46:29.5694295Z 
2020-09-27T07:46:29.7422355Z *** Cleanup VSTS Environment: True
2020-09-27T07:46:32.2729455Z ##[section]Finishing: ALOps Extension API
```

**Expected behavior**
I would've expected the ALOps Extension API step to report status: Error 

**Screenshots**
![image](https://user-images.githubusercontent.com/968411/94363745-c3976100-00c4-11eb-98d6-2961115e6aa6.png)
![image](https://user-images.githubusercontent.com/968411/94363748-c85c1500-00c4-11eb-9da0-558dfa5fe156.png)

**Comment by waldo1001** [comment]
It's the way DevOps reports it back (depending on the agents) - we are aware.

We have a new release on the shelf (going through test-phase now) ...

**Comment by AdminHodor** [comment]
Dear @JeppeBylov ,

We just released v1.435.1856 which should tackle this issue.
Please note it could take a couple of pipeline runs for DevOps to pickup the new version.

Kind regards,
